libs/agno/agno/agent/agent.py

```diff
@@ -1081,13 +1106,32 @@ async def _arun(
         # 2. Generate a response from the Model (includes running function calls)
         model_response: ModelResponse = await self.model.aresponse(
             messages=run_messages.messages,
-            response_format=response_format,
             tools=self._tools_for_model,
             functions=self._functions_for_model,
             tool_choice=self.tool_choice,
             tool_call_limit=self.tool_call_limit,
+            **({"response_format": response_format} if self.parser_model is None else {}),
         )
 
+        # If a parser model is provided, structure the response separately
+        if self.parser_model is not None:
+            messages_for_parser_model = self.get_messages_for_parser_model(model_response)
+            parser_model_response: ModelResponse = await self.parser_model.aresponse(
+                messages=messages_for_parser_model,
+                response_format=self.response_model,
+            )
+            parser_model_response_message: Optional[Message] = None
+            for message in reversed(messages_for_parser_model):
+                if message.role == "assistant":
+                    parser_model_response_message = message
+                    break
+            if parser_model_response_message is not None:
+                run_messages.messages.append(parser_model_response_message)
```

Comment:
Should it not replace the last assistant message? Worried it will mess with things if there are 2 messages.